erpnext/manufacturing/doctype/production_plan/production_plan.js

```diff
@@ -32,6 +32,10 @@ frappe.ui.form.on('Production Plan', {
 		}
 	},
 
+	validate: function(frm ) {
```

Comment:
@MyuddinKhatri this will give a mandatory field error. so your message regarding default bom not able appear any way.
what is the use case  if you want to save the document without BOM then it should check bom_madetory before submit